src/ui/pages/Contract.tsx

```diff
@@ -73,22 +75,29 @@ export function Contract() {
   const projectName = contract?.abi.info.contract.name;
 
   return (
-    <Loader isLoading={!contract && isLoading}>
+    <Loader isLoading={(!contract && isLoading) || isNull(isOnChain)}>
```

Comment:
why should a contract that is not on-chain trigger a loading state for the page? why do we need a loader here anyway? it was probably meant to be for remote fetching but that is not implemented and probably never will be